tests/rptest/services/redpanda.py

```diff
@@ -1586,6 +1590,19 @@ def stop(self, **kwargs):
                         lambda n: self.stop_node(n, **kwargs),
                         parallel=True)
 
+        if len(self.failure_injectors) > 0:
+            for ident, injector_wref in self.failure_injectors.items():
+                if injector_wref:
+                    self.logger.error(
+                        f"Test left behind live failure injector with identifier: {ident}."
```

Comment:
I think we would be better off using the `Service` infrastructure in ducktape: this already keeps a registry of instantiated Services and makes sure they get their stop/clean methods called at the end of a test, even if the test doesn't remember to do it.  We could make the failure injector class inherit from Service and just not acquire any nodes.

Reply:
Not opposed to the idea, but I find it weird to have a service with no nodes. Perhaps it's a perfectly normal think to do in ducktape though.